Description: Bump the self-test timeouts to increase autopkgtest success rate
Especially on i386 tests the systemd selftests were flaky for quite a while.
It turned out that 5/8 tests checked seemed to have worked fine but were
killed early by the timeouts expiring.
It was brought up that spectre and L1TF mitigations might have further
opened the window for these issues to trigger more often now.
Lets in our package bump the timeout which will worst case make a real bad test
slightly longer but probably safes many hours of wasted tests especially
considering how often they are jsut retried these days.
.
We might forward that upstream if for a while this proves to increase
the success rate of systemd autopkgtests.
Forwarded: no
Forward-info: need to prove with test success rate
Author: Christian Ehrhardt <christian.ehrhardt@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1789841
Last-Update: 2018-08-30
--- a/test/TEST-08-ISSUE-2730/test.sh
+++ b/test/TEST-08-ISSUE-2730/test.sh
@@ -6,7 +6,7 @@ TEST_DESCRIPTION="https://github.com/sys
 TEST_NO_NSPAWN=1
 
 . $TEST_BASE_DIR/test-functions
-QEMU_TIMEOUT=180
+QEMU_TIMEOUT=300
 FSTYPE=ext4
 
 test_setup() {
--- a/test/TEST-09-ISSUE-2691/test.sh
+++ b/test/TEST-09-ISSUE-2691/test.sh
@@ -6,7 +6,7 @@ TEST_DESCRIPTION="https://github.com/sys
 TEST_NO_NSPAWN=1
 
 . $TEST_BASE_DIR/test-functions
-QEMU_TIMEOUT=90
+QEMU_TIMEOUT=300
 
 test_setup() {
     create_empty_image
--- a/test/TEST-17-UDEV-WANTS/test.sh
+++ b/test/TEST-17-UDEV-WANTS/test.sh
@@ -6,7 +6,7 @@ TEST_DESCRIPTION="UDEV SYSTEMD_WANTS pro
 TEST_NO_NSPAWN=1
 
 . $TEST_BASE_DIR/test-functions
-QEMU_TIMEOUT=180
+QEMU_TIMEOUT=300
 
 test_setup() {
     create_empty_image
--- a/test/TEST-18-FAILUREACTION/test.sh
+++ b/test/TEST-18-FAILUREACTION/test.sh
@@ -5,7 +5,7 @@ set -e
 TEST_DESCRIPTION="FailureAction= operation"
 
 . $TEST_BASE_DIR/test-functions
-QEMU_TIMEOUT=180
+QEMU_TIMEOUT=300
 
 test_setup() {
     create_empty_image
--- a/test/TEST-19-DELEGATE/test.sh
+++ b/test/TEST-19-DELEGATE/test.sh
@@ -6,7 +6,7 @@ TEST_DESCRIPTION="test cgroup delegation
 TEST_NO_NSPAWN=1
 
 . $TEST_BASE_DIR/test-functions
-QEMU_TIMEOUT=180
+QEMU_TIMEOUT=300
 UNIFIED_CGROUP_HIERARCHY=yes
 
 test_setup() {
